electrum/lnonion.py

```diff
@@ -211,6 +259,29 @@ def new_onion_packet(
         hmac=next_hmac)
 
 
+def encrypt_encrypted_data_tlv(*, shared_secret, **kwargs):
```

Comment:
This name is unfortunate. It suggests that we are encrypting a second time something that was already encrypted. It is enough to call this method `encrypt_data_tlv` (and to call the next one `decrypt_data_tlv`). And, yes, I am aware that the name `encrypted_data_tlv` is part of the spec. That's not a reason.

Reply:
Agreed, this looks confusing. I would suggest `encrypt_onionmsg_data_tlv` then, to indicate it is an onion message record.